build_tools/circle/push_doc.sh

```diff
@@ -24,9 +24,11 @@ MSG="Pushing the docs to $dir/ for branch: $CIRCLE_BRANCH, commit $CIRCLE_SHA1"
 
 cd $HOME
 if [ ! -d $DOC_REPO ];
-then git clone "git@github.com:scikit-learn/"$DOC_REPO".git";
+then git clone --depth 1 --no-checkout "git@github.com:scikit-learn/"$DOC_REPO".git";
 fi
 cd $DOC_REPO
+git config core.sparseCheckout true
+echo $dir > .git/info/sparse-checkout
```

Comment:
I am really surprised that this should be done manually. It seems very error prone.

Are you sure that this is the right way to do a sparse checkout?